File: backend/app/schemas/application.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.models.application import ApplicationStatus


class ApplicationIn(BaseModel):
    company: str
    position: str
    job_url: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    status: ApplicationStatus = ApplicationStatus.APPLIED
    applied_date: date | None = None
    notes: str | None = None


class ApplicationOut(ApplicationIn):
    model_config = {"from_attributes": True}
    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/application.ts
export type ApplicationStatus =
  | 'APPLIED'
  | 'PHONE_SCREEN'
  | 'INTERVIEW'
  | 'OFFER'
  | 'REJECTED'
  | 'WITHDRAWN'

export const APPLICATION_STATUSES: ApplicationStatus[] = [
  'APPLIED',
  'PHONE_SCREEN',
  'INTERVIEW',
  'OFFER',
  'REJECTED',
  'WITHDRAWN',
]

export const STATUS_LABELS: Record<ApplicationStatus, string> = {
  APPLIED: 'Applied',
  PHONE_SCREEN: 'Phone Screen',
  INTERVIEW: 'Interview',
  OFFER: 'Offer',
  REJECTED: 'Rejected',
  WITHDRAWN: 'Withdrawn',
}

export const STATUS_COLORS: Record<ApplicationStatus, string> = {
  APPLIED: '#3b82f6',
  PHONE_SCREEN: '#8b5cf6',
  INTERVIEW: '#f59e0b',
  OFFER: '#10b981',
  REJECTED: '#ef4444',
  WITHDRAWN: '#6b7280',
}

export interface JobApplication {
  id: string
  user_id: string
  company: string
  position: string
  job_url: string | null
  salary_min: number | null
  salary_max: number | null
  status: ApplicationStatus
  applied_date: string | null
  notes: string | null
  created_at: string
  updated_at: string
}

export type ApplicationIn = Omit<JobApplication, 'id' | 'user_id' | 'created_at' | 'updated_at'>
